features/generator_features.py

```diff
@@ -208,13 +226,18 @@ def _start_wait_retry(self):
         self.perturbation_rotation_z = self.init_rotation_z + self.delta_rotation_z*num_deltas
         self.perturbation_rotation_x = self.init_rotation_x + self.delta_rotation_x*num_deltas
 
+        # change tracking out time of final rotation
+        if num_deltas+1 == self.num_increments:
+            self.tracking_out_time = self.final_tracking_out_time
+
         # stop incrementing once final perturbation rotation reached
         if self.num_trials_success >= self.num_increments * self.trials_per_increment:
             self.pertubation_rotation = self.final_rotation_y
             self.perturbation_rotation_z = self.final_rotation_z
             self.perturbation_rotation_x = self.final_rotation_x
         
-        print(self.pertubation_rotation)
+        print("Y", self.pertubation_rotation, "Z", self.perturbation_rotation_z, "X", self.perturbation_rotation_x)
+        print(self.tracking_out_time, "tracking out")
 
```

Comment:
seems like _start_wait_retry() and _start_wait() are identical to me, could you call _start_wait() from _start_wait_retry() instead of copying all these lines of code? same applies to the _start_wait() and _start_wait_retry() in target_tracking_task.py

Reply:
I made this change in both the IncrementalRotation feature and target_tracking_task.py!